### PR TITLE
refactor: int constants => enum

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Those commits add some enum for the sake of code clarity

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -2672,19 +2674,24 @@ class NoteEditor : AnkiFragment(R.layout.note_editor), DeckSelectionListener, Su
...
-        const val CALLER_NO_CALLER = 0
-        const val CALLER_EDIT = 1
-        const val CALLER_STUDYOPTIONS = 2
-        const val CALLER_DECKPICKER = 3
-        const val CALLER_REVIEWER_ADD = 11
-        const val CALLER_CARDBROWSER_ADD = 7
-        const val CALLER_NOTEEDITOR = 8
-        const val CALLER_PREVIEWER_EDIT = 9
-        const val CALLER_NOTEEDITOR_INTENT_ADD = 10
+        enum class Caller {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:2677]: call it `NoteEditorCaller`

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 2]
```diff
@@ -2672,19 +2674,24 @@ class NoteEditor : AnkiFragment(R.layout.note_editor), DeckSelectionListener, Su
...
+            REVIEWER_ADD,
+            CARDBROWSER_ADD,
+            NOTEEDITOR,
+            PREVIEWER_EDIT,
+            NOTEEDITOR_INTENT_ADD,
+            IMG_OCCLUSION,
+            ADD_IMAGE,
+            INSTANT_NOTE_EDITOR;
+            companion object {
+                fun Int.toCaller() = Caller.entries[this]
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:2691]: This is a recipe for disaster, keep the enum ints explicit

This now throws on an unexpected value

**Review by david-allison — CHANGES_REQUESTED** [review]
Note: both of these classes are planned to be deleted.

Signal is fine. Good change

I don't like `Caller` from an architecture POV, but I don't feel strongly here. The code's going to go away anyway.

I do want to stop using implicit enum ordinals, and that's what I'll request changes on: that makes the order of the enums a source of bugs.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 3]
```diff
@@ -2672,19 +2674,24 @@ class NoteEditor : AnkiFragment(R.layout.note_editor), DeckSelectionListener, Su
...
+            NOTEEDITOR(8),
+            PREVIEWER_EDIT(9),
+            NOTEEDITOR_INTENT_ADD(10),
+            IMG_OCCLUSION(12),
+            ADD_IMAGE(13),
+            INSTANT_NOTE_EDITOR(14);
+            companion object {
+                fun Int.toCaller() = NoteEditorCaller.entries.first { it.value == this }
+            }
+        }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:2693]: why is `11` not in the enum?

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:2693]: they are out of order, it's there, in line 2682 

https://github.com/ankidroid/Anki-Android/pull/17435/files#diff-6d9f7ea7bba9cc72513b5078633cee9bdbbd1ea7646114869679f302ad607459R2682

they were out of order to start so this is actually the most understandable diff and is likely why Arthur just left it in that order (but I'm just guessing there of course)

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:2693]: Thanks! That took me a while, as the constant on the next line was also 11

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
fine by me, assuming conflict is cleaned up merge at will IMHO

**Pull request merged**